util/featureflag/featureflag_test.go

```diff
@@ -54,39 +55,48 @@ func Test_VersionGuard(t *testing.T) {
 	e := gin.Default()
 	e.Use(f1.VersionGuard())
 	e.GET("/ping", func(c *gin.Context) {
-		c.String(200, "pong")
+		c.String(http.StatusOK, "pong")
 	})
 	w := httptest.NewRecorder()
 	req, _ := http.NewRequest("GET", "/ping", nil)
 	e.ServeHTTP(w, req)
 
-	r.Equal(200, w.Code)
+	r.Equal(http.StatusOK, w.Code)
 	r.Equal("pong", w.Body.String())
 
-	// abort with other middlewares
+	// abort
 	handled := false
 	e2 := gin.Default()
 	e2.Use(func(c *gin.Context) {
 		c.Next()
 
 		handled = true
-
-		// check error type
-		currentErr := c.Errors.Last().Err
-		codeProperty, _ := rest.HTTPCodeProperty(http.StatusForbidden)
-		code, ok := currentErr.(*errorx.Error).Property(codeProperty)
-		r.True(ok)
-
-		r.Equal(true, errorx.IsOfType(currentErr, ErrFeatureUnsupported))
-		r.Equal(http.StatusForbidden, code)
+		r.Equal(true, errorx.IsOfType(c.Errors.Last().Err, ErrFeatureUnsupported))
 	})
 	e2.Use(f1.VersionGuard(), f2.VersionGuard())
 	e2.GET("/ping", func(c *gin.Context) {
-		c.String(200, "pong")
+		c.String(http.StatusOK, "pong")
 	})
 	w2 := httptest.NewRecorder()
 	req2, _ := http.NewRequest("GET", "/ping", nil)
 	e2.ServeHTTP(w2, req2)
 
 	r.Equal(true, handled)
 }
+
+func Test_VersionGuardWith_ErrorHandlerFn(t *testing.T) {
+	r := require.New(t)
+	f := newFeatureFlag("testFeature", "v5.3.0", ">= 5.3.1")
+
+	e := gin.Default()
+	e.Use(rest.ErrorHandlerFn())
+	e.Use(f.VersionGuard())
+	e.GET("/ping", func(c *gin.Context) {
+		c.String(http.StatusOK, "pong")
+	})
+	w2 := httptest.NewRecorder()
+	req2, _ := http.NewRequest("GET", "/ping", nil)
+	e.ServeHTTP(w2, req2)
+
+	r.Equal(http.StatusForbidden, w2.Code)
```

Comment:
Forbidden means Unauthorized: The HTTP 403 Forbidden response status code indicates that the server understands the request but refuses to authorize it.

However there is no way for this router to be authorized. So maybe BadRequest is better.